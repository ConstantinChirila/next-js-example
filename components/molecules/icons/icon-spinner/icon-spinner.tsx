import React, { ReactElement } from 'react'
import {
  SVGIcon,
  TIconProperties,
  VisuallyHidden,
} from '@tooploox-test/components'

export function IconSpinner(properties: TIconProperties): ReactElement {
  return (
    <SVGIcon
      viewBox="0 0 100 100"
      enable-background="new 0 0 0 0"
      {...properties}
      aria-hidden="true"
    >
      <VisuallyHidden as="title">Spinner icon</VisuallyHidden>
      <circle fill="none" strokeWidth="4" cx="50" cy="50" r="44" />
      <circle strokeWidth="3" cx="8" cy="54" r="6">
        <animateTransform
          attributeName="transform"
          dur="2s"
          type="rotate"
          from="0 50 48"
          to="360 50 52"
          repeatCount="indefinite"
        />
      </circle>
    </SVGIcon>
  )
}
