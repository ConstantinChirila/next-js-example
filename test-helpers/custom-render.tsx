import { PropsWithChildren, ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'
import { render, RenderResult, RenderOptions } from '@testing-library/react'
import { QueryClientProvider, QueryClient } from 'react-query'

import { defaultTheme } from '@tooploox-test/theme'
import { reactQueryDefaults } from '@tooploox-test/configuration'

function DefaultThemeProviderWithReactQuery({
  children,
}: PropsWithChildren<{}>): ReactElement {
  const queryClient = new QueryClient({
    ...reactQueryDefaults,
  })

  return (
    <ThemeProvider theme={defaultTheme}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  )
}

function DefaultThemeProvider({
  children,
}: PropsWithChildren<{}>): ReactElement {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
}

function renderWithReactQueryContext(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProviderWithReactQuery, ...options })
}

function renderWithTheme(
  ui: ReactElement,
  options?: RenderOptions
): RenderResult {
  return render(ui, { wrapper: DefaultThemeProvider, ...options })
}

export { renderWithTheme, renderWithReactQueryContext }
