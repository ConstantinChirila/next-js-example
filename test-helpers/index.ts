export * from './custom-render'
export * from './setup'
