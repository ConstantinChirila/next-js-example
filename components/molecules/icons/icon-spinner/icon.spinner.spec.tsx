import React from 'react'

import { IconSpinner } from './icon-spinner'
import { renderWithTheme } from '../../../../test-helpers'

describe('IconSpinner', () => {
  it('should match a snapshot', () => {
    const { container } = renderWithTheme(<IconSpinner />)
    expect(container).toMatchSnapshot()
  })
})
