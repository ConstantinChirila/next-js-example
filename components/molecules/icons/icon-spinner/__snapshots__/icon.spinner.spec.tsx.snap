// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconSpinner should match a snapshot 1`] = `
.c0 {
  display: block;
  height: 100%;
  width: 100%;
}

.c1 {
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 0.1rem;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

<div>
  <svg
    aria-hidden="true"
    class="c0"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      class="c1"
    >
      Spinner icon
    </title>
    <circle
      cx="50"
      cy="50"
      fill="none"
      r="44"
      stroke-width="4"
    />
    <circle
      cx="8"
      cy="54"
      r="6"
      stroke-width="3"
    >
      <animatetransform
        attributeName="transform"
        dur="2s"
        from="0 50 48"
        repeatCount="indefinite"
        to="360 50 52"
        type="rotate"
      />
    </circle>
  </svg>
</div>
`;
